### ISSUE TITLE
crash passing a function as attr.int's default

### ISSUE THREAD

**Issue by benjaminp** [bazelbuild/bazel#9463]
```
$ bazel version
Build label: 0.29.1
Build target: bazel-out/k8-opt/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Tue Sep 10 13:44:39 2019 (1568123079)
Build timestamp: 1568123079
Build timestamp as int: 1568123079
$ touch WORKSPACE
$ cat > test.bzl
def _impl(ctx): pass

r = rule(implementation = _impl, attrs = {'i': attr.int(default = _impl)})
$ cat > BUILD
load(':test.bzl', 'r')
$ bazel build *
Internal error thrown during build. Printing stack trace: java.lang.RuntimeException: Unrecoverable error while evaluating node '//:test.bzl' (requested by nodes 'PACKAGE:')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:528)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:399)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: argument type mismatch
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.devtools.build.lib.syntax.MethodDescriptor.call(MethodDescriptor.java:135)
	at com.google.devtools.build.lib.syntax.FuncallExpression.invokeObjectMethod(FuncallExpression.java:930)
	at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:898)
	at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
	at com.google.devtools.build.lib.syntax.DictionaryLiteral.doEval(DictionaryLiteral.java:73)
	at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
	at com.google.devtools.build.lib.syntax.FuncallExpression.evalArguments(FuncallExpression.java:839)
	at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:903)
	at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
	at com.google.devtools.build.lib.syntax.Eval.execAssignment(Eval.java:72)
	at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:192)
	at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
	at com.google.devtools.build.lib.syntax.BuildFileAST.execTopLevelStatement(BuildFileAST.java:266)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.execAndExport(SkylarkImportLookupFunction.java:558)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.createExtension(SkylarkImportLookupFunction.java:539)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.computeInternal(SkylarkImportLookupFunction.java:395)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.compute(SkylarkImportLookupFunction.java:103)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:451)
	... 4 more

INFO: Elapsed time: 0.380s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (1 packages loaded)
    currently loading: 
Internal error thrown during build. Printing stack trace: java.lang.RuntimeException: Unrecoverable error while evaluating node '//:test.bzl' (requested by nodes 'PACKAGE:')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:528)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:399)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: argument type mismatch
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.devtools.build.lib.syntax.MethodDescriptor.call(MethodDescriptor.java:135)
	at com.google.devtools.build.lib.syntax.FuncallExpression.invokeObjectMethod(FuncallExpression.java:930)
	at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:898)
	at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
	at com.google.devtools.build.lib.syntax.DictionaryLiteral.doEval(DictionaryLiteral.java:73)
	at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
	at com.google.devtools.build.lib.syntax.FuncallExpression.evalArguments(FuncallExpression.java:839)
	at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:903)
	at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
	at com.google.devtools.build.lib.syntax.Eval.execAssignment(Eval.java:72)
	at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:192)
	at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
	at com.google.devtools.build.lib.syntax.BuildFileAST.execTopLevelStatement(BuildFileAST.java:266)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.execAndExport(SkylarkImportLookupFunction.java:558)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.createExtension(SkylarkImportLookupFunction.java:539)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.computeInternal(SkylarkImportLookupFunction.java:395)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.compute(SkylarkImportLookupFunction.java:103)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:451)
	... 4 more
java.lang.RuntimeException: Unrecoverable error while evaluating node '//:test.bzl' (requested by nodes 'PACKAGE:')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:528)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:399)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: argument type mismatch
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.devtools.build.lib.syntax.MethodDescriptor.call(MethodDescriptor.java:135)
	at com.google.devtools.build.lib.syntax.FuncallExpression.invokeObjectMethod(FuncallExpression.java:930)
	at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:898)
	at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
	at com.google.devtools.build.lib.syntax.DictionaryLiteral.doEval(DictionaryLiteral.java:73)
	at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
	at com.google.devtools.build.lib.syntax.FuncallExpression.evalArguments(FuncallExpression.java:839)
	at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:903)
	at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
	at com.google.devtools.build.lib.syntax.Eval.execAssignment(Eval.java:72)
	at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:192)
	at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
	at com.google.devtools.build.lib.syntax.BuildFileAST.execTopLevelStatement(BuildFileAST.java:266)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.execAndExport(SkylarkImportLookupFunction.java:558)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.createExtension(SkylarkImportLookupFunction.java:539)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.computeInternal(SkylarkImportLookupFunction.java:395)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.compute(SkylarkImportLookupFunction.java:103)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:451)
	... 4 more
java.lang.RuntimeException: Unrecoverable error while evaluating node '//:test.bzl' (requested by nodes 'PACKAGE:')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:528)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:399)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.IllegalArgumentException: argument type mismatch
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.devtools.build.lib.syntax.MethodDescriptor.call(MethodDescriptor.java:135)
	at com.google.devtools.build.lib.syntax.FuncallExpression.invokeObjectMethod(FuncallExpression.java:930)
	at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:898)
	at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
	at com.google.devtools.build.lib.syntax.DictionaryLiteral.doEval(DictionaryLiteral.java:73)
	at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
	at com.google.devtools.build.lib.syntax.FuncallExpression.evalArguments(FuncallExpression.java:839)
	at com.google.devtools.build.lib.syntax.FuncallExpression.doEval(FuncallExpression.java:903)
	at com.google.devtools.build.lib.syntax.Expression.eval(Expression.java:75)
	at com.google.devtools.build.lib.syntax.Eval.execAssignment(Eval.java:72)
	at com.google.devtools.build.lib.syntax.Eval.execDispatch(Eval.java:192)
	at com.google.devtools.build.lib.syntax.Eval.exec(Eval.java:183)
	at com.google.devtools.build.lib.syntax.BuildFileAST.execTopLevelStatement(BuildFileAST.java:266)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.execAndExport(SkylarkImportLookupFunction.java:558)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.createExtension(SkylarkImportLookupFunction.java:539)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.computeInternal(SkylarkImportLookupFunction.java:395)
	at com.google.devtools.build.lib.skyframe.SkylarkImportLookupFunction.compute(SkylarkImportLookupFunction.java:103)
FAILED: Build did NOT complete successfully (1 packages loaded)
    currently loading: 
```

**Comment by laurentlb** [comment]
@alandonovan - can you take a look at this crash?

**Comment by alandonovan** [comment]
I can reproduce it; looking into a fix now. Sorry for the breakage.

**Comment by benjaminp** [comment]
I don't think this is new breakage. It looks to me that `attr.int` has had the wrong annotation since 0446f99c4b0f9739aca53826ae3e32555fcf47a6.

**Comment by alandonovan** [comment]
You're absolutely right; I've been making many changes to lib/syntax recently and this looked like plausible fallout from one of them. I have a fix pending.

**Comment by alandonovan** [comment]
Fixed by internal CL 286037108.

**Comment by jin** [comment]
Closing as reported fix.
